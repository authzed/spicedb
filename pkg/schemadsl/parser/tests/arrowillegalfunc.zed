definition resource {
    permission view = a.foo(bar)
}