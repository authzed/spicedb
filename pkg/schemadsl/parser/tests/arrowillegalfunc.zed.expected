NodeTypeFile
  end-rune = 48
  input-source = arrow illegal function test
  start-rune = 0
  child-node =>
    NodeTypeDefinition
      definition-name = resource
      end-rune = 48
      input-source = arrow illegal function test
      start-rune = 0
      child-node =>
        NodeTypePermission
          end-rune = 48
          input-source = arrow illegal function test
          relation-name = view
          start-rune = 26
          child-node =>
            NodeTypeError
              end-rune = 48
              error-message = Expected 'any' or 'all' for arrow function, found: foo
              error-source = (
              input-source = arrow illegal function test
              start-rune = 49
            NodeTypeError
              end-rune = 48
              error-message = Expected right hand expression, found: TokenTypeLeftParen
              error-source = (
              input-source = arrow illegal function test
              start-rune = 49
          compute-expression =>
            NodeTypeIdentifier
              end-rune = 44
              identifier-value = a
              input-source = arrow illegal function test
              start-rune = 44
        NodeTypeError
          end-rune = 48
          error-message = Expected end of statement or definition, found: TokenTypeLeftParen
          error-source = (
          input-source = arrow illegal function test
          start-rune = 49
    NodeTypeError
      end-rune = 48
      error-message = Unexpected token at root level: TokenTypeLeftParen
      error-source = (
      input-source = arrow illegal function test
      start-rune = 49